src/atomate2/vasp/flows/phonons.py

```diff
@@ -133,3 +133,15 @@ class PhononMaker(BasePhononMaker):
     phonon_displacement_maker: BaseVaspMaker = field(
         default_factory=PhononDisplacementMaker
     )
+
+    @property
+    def prev_calc_dir_argname(self):
```

Comment:
The same issue, should we add `@abstractmethod` decoration here as it is in [src/atomate2/common/flows/phonons.py](url)?

Reply:
I don't think we need the `@abstractmethod` decorator there since in the child classes those are not abstract. `@abstractmethod` tells python that a method  will be overwritten by the child class.

Reply:
If someone were to make `PhononMakerVaspSpecialFeature` inheriting from `PhononMakerVasp` then `prev_calc_dir_argname` could become abstract in `PhononMakerVasp` if `PhononMakerVaspSpecialFeature` were to change that property.

Reply:
okay, just want to confirm it is only needed in some cases.
@tpurcell90 Thanks for the point-by-points fixes. It is very clear and easy to track your commits.
I suppose it is ready to merge now.
Since it is such a heavy PR, I would like to wait for @utf to click the final button.

Reply:
The only two open things are the ASE dependency linking to gitlab vs an exact version (waiting for the new release) and that I want my colleague/coauthor @ansobolev of the code base to do one final review to make sure that he agrees to it.

Reply:
Just wanted to add that I mainly looked at the phonon parts in the review. Thus, reviewing the other parts and that they agree with atomate2 as well, might be good.

I had one additional question: do the units from the output schema agree with the VASP units? E.g., if we wanted to build a common elastic workflow, would we need to convert? I think we have such a conversion for the forcefields. (@utf, what overall philosophy does atomate2 follow here?)

Reply:
@ansobolev can correct me, but I think they are the same with the caveat that FHI-aims uses CODATA 2002 for all of its units so we may differ at the 6-9th digit after the decimal.

Reply:
For clarification internally FHI-aims uses Hartrees and then converts to eV using CODATA 2002 constants, which can lead to small differences

Reply:
Thanks!

Stresses as well? I just remember a conversion that was necessary for the stresses coming from the forcefields in ase. Maybe, I am mistaken. Just wamt to make sure.

Reply:
What units does VASP use for stresses? FHI-aims uses ev/A^3, but I think there was a discussion at some point about a CODATA 2002 discrepancy between vibes and FHI-aims. I forget the issue, but that is when we learned ASE uses the modern CODATA and FHI-aims uses the 2002 version and that led to issues at the 6-9th decimal.

Reply:
It's kbar (see https://www.vasp.at/wiki/index.php/PSTRESS): 
https://github.com/materialsproject/atomate2/blob/afcce840c7a86be527efbfc09c2b0864906f50a5/src/atomate2/forcefields/schemas.py#L59

Reply:
Okay we can do the same conversion in our schema as well so they are all consistent withe each other

Reply:
Yes. And let's wait for @utf 's suggestions :).

Reply:
I made the change in the parser, but that can be easily reverted (one multiplication factor). I also noticed that I copied over one too many tests for the parser so I removed it